File: Missing Number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int xor1 = 0, xor2 = 0;
        int n = nums.size(); // Total number of elements (including the missing one)

        // XOR of array elements
        for (int i = 0; i < n; i++) {
            xor2 ^= nums[i];
        }

        // XOR of numbers from 0 to n
        for (int i = 0; i <= n; i++) {
            xor1 ^= i;
        }

        // The missing number is the XOR of the above two results
        return xor1 ^ xor2;
    }
};

File: Missing Number.java
class Solution {
    int missingNum(int arr[]) {
        // code here
         int n = arr.length + 1;  // Since one number is missing
        int total = n * (n + 1) / 2;

        int sum = 0;
        for (int num : arr) {
            sum += num;
        }

     return (int)(total - sum);
    }
}
